app/models/user.rb

```diff
@@ -58,6 +58,25 @@ class User < ApplicationRecord
 
   scope :with_same_userid, ->(id) { where(:userid => User.where(:id => id).pluck(:userid)) }
 
+  # This just creates the following Hash:
+  #
+  #   {
+  #     :api_includes => :api_includes
+  #   }
+  #
+  # Where :default_scoped is the value for anything that doesn't match a key in
+  # the original array.  The "lookup hash" was chossen since it is much faster
+  # to do a hash table lookup than a `Array.includes?(KEY)`.
+  LOOKUP_SCOPES = [:api_includes].index_by(&:itself) .reverse_merge(Hash.new(:default_scoped)) .freeze
```

Comment:
why `[:api_includes].index_by(&:itself)` vs `{:api_includes => :api_includes}`?

you could delete the 6 line comment, make the code shorter, and more obvious.

Feel like you had a reason for this but can't figure it out.

Reply:
My thought was that if we wanted to add to it, it was just as simple as adding it once instead of twice (once in the area instead of twice in the hash).

That said, the "cleverness" might not be worth it.

Reply:
Fixed